pkg/clusteragent/admission/mutate/agent_sidecar/util.go

```diff
@@ -116,3 +117,79 @@ func applyFargateOverrides(container *corev1.Container) {
 		Value: "true",
 	})
 }
+
+////////////////////////////////
+//                            //
+//     Profiles Overrides     //
+//                            //
+////////////////////////////////
+
+// ProfileOverride represents environment variables and resource requirements overrides
+type ProfileOverride struct {
+	EnvVars              []corev1.EnvVar             `json:"env,omitempty"`
+	ResourceRequirements corev1.ResourceRequirements `json:"resources,omitempty"`
+}
+
+func applyProfileOverrides(container *corev1.Container) {
+	if container == nil {
+		return
+	}
+
+	// Read and parse profiles
+	profilesJSON := config.Datadog.GetString("admission_controller.agent_sidecar.profiles")
+
+	var profiles []ProfileOverride
+
+	err := json.Unmarshal([]byte(profilesJSON), &profiles)
+	if err != nil {
+		log.Errorf("failed to parse profiles for admission controller agent sidecar injection: %s", err)
+		return
+	}
+
+	if len(profiles) > 1 {
+		log.Errorf("only 1 profile is supported")
+		return
+	}
+
+	if len(profiles) == 0 {
+		return
+	}
+
+	overrides := profiles[0]
+
+	// Apply environment variable overrides
+	for _, envVarOverride := range overrides.EnvVars {
+		// Check if the environment variable already exists in the container
+		var found bool
+		for i, envVar := range container.Env {
+			if envVar.Name == envVarOverride.Name {
+				// Override the existing environment variable value
+				container.Env[i] = envVarOverride
+				found = true
+				break
+			}
+		}
+		// If the environment variable doesn't exist, add it to the container
+		if !found {
+			container.Env = append(container.Env, envVarOverride)
+		}
+	}
```

Comment:
**💭 thought:** ‏I think this is the same code as `withEnvOverrides`. If future changes on this part of the code also need to be applied to `withEnvOverrides` I believe it's valuable to factorize it.

Reply:
`withEnvOverrides` is created as a util function for unit tests only.

But I agree that it can be good to factorize the code and have the logic implemented in one place.

Will do this.

thanks for the suggestion 🙇‍♂️